api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -315,6 +323,10 @@ private void setDbWorkspaceFields(
     if (Strings.isNullOrEmpty(workspace.getEtag())) {
       throw new BadRequestException("Missing required update field 'etag'");
     }
+
+    DbWorkspace dbWorkspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
```

Comment:
OK, functional changes are definitely not intended here.

Reply:
I was pointing it out as a somewhat derived security hole.

Annotating that method is not a bad idea. I definitely think there's an issue where it's easy to make unintended changes since a number of our calls have side effects that are not immediately apparent. I think there are a few ideas we should explore for the long term though. Some off the top of my head are better function names or using checked exceptions.